Code/GraphMol/Depictor/EmbeddedFrag.cpp

```diff
@@ -346,6 +346,76 @@ void EmbeddedFrag::setupAttachmentPoints() {
   }
 }
 
+// check if the stereochemistry of the template matches the stereochemistry of
+// the molecule
+static bool checkStereoChemistry(const RDKit::ROMol &mol,
+                                 const RDKit::ROMol &template_mol,
+                                 RDKit::MatchVectType match) {
+  std::cerr << "checkStereoChemistry" << std::endl;
+  for (auto bond : mol.bonds()) {
+    if (bond->getBondType() != RDKit::Bond::DOUBLE ||
+        bond->getStereo() == RDKit::Bond::STEREOANY ||
+        bond->getStereo() == RDKit::Bond::STEREONONE) {
+      continue;
+    }
+    // get the four atoms around the double bond
+    auto neighbors = bond->getStereoAtoms();
+    if (neighbors.size() != 2) {
+      continue;
+    }
+    int atom1_neighbor = neighbors[0];
+    int atom2_neighbor = neighbors[1];
+    int atom1 = bond->getBeginAtom()->getIdx();
+    int atom2 = bond->getEndAtom()->getIdx();
+    // find the template atoms that correspond to the four atoms
+    int template_atom1 = -1;
+    int template_atom2 = -1;
+    int template_atom1_neighbor = -1;
+    int template_atom2_neighbor = -1;
+    for (auto &[template_aidx, rs_aidx] : match) {
+      if (rs_aidx == atom1) {
+        template_atom1 = template_aidx;
+      } else if (rs_aidx == atom2) {
+        template_atom2 = template_aidx;
+      } else if (rs_aidx == atom1_neighbor) {
+        template_atom1_neighbor = template_aidx;
+      } else if (rs_aidx == atom2_neighbor) {
+        template_atom2_neighbor = template_aidx;
+      }
+    }
+    if (template_atom1 == -1 || template_atom2 == -1 ||
+        template_atom1_neighbor == -1 || template_atom2_neighbor == -1) {
+      return false;
+    }
+    auto conf = template_mol.getConformer();
+    RDGeom::Point2D atom1_loc(conf.getAtomPos(template_atom1));
+    RDGeom::Point2D atom2_loc(conf.getAtomPos(template_atom2));
+    RDGeom::Point2D atom1_neighbor_loc(
+        conf.getAtomPos(template_atom1_neighbor));
+    RDGeom::Point2D atom2_neighbor_loc(
+        conf.getAtomPos(template_atom2_neighbor));
+    // check if the two neighbors are on the same side of the bond
+    auto x1 = atom1_neighbor_loc.x;
+    auto y1 = atom1_neighbor_loc.y;
+    auto x2 = atom1_loc.x;
+    auto y2 = atom1_loc.y;
+    auto x3 = atom2_loc.x;
+    auto y3 = atom2_loc.y;
+    auto x4 = atom2_neighbor_loc.x;
+    auto y4 = atom2_neighbor_loc.y;
+    auto cross1 = (x3 - x2) * (y1 - y2) - (y3 - y2) * (x1 - x2);
+    auto cross2 = (x3 - x2) * (y4 - y2) - (y3 - y2) * (x4 - x2);
```

Comment:
This is more efficient and, IMO, easier to read.
```suggestion
    const auto &conf = template_mol.getConformer();
    const auto &atom1_loc = conf.getAtomPos(template_atom1);
    const auto &atom2_loc = conf.getAtomPos(template_atom2);
    const auto &atom1_neighbor_loc = conf.getAtomPos(template_atom1_neighbor);
    const auto &atom2_neighbor_loc = conf.getAtomPos(template_atom2_neighbor);
    // check if the two neighbors are on the same side of the bond
    const auto v12 = atom1_neighbor_loc - atom1_loc;
    const auto v42 = atom2_neighbor_loc - atom1_loc;
    const auto v32 = atom2_loc - atom1_loc;
    auto cross1 = v32.x * v12.y - v32.y * v12.x;
    auto cross2 = v32.x * v42.y - v32.y * v42.x;
```